### ISSUE TITLE
Create parameterized queries

### ISSUE THREAD

**Issue by VahidN** [alirezanet/Gridify#24]
This library in-lines the field's values in EF-Core queries. For instance it creates queries like this:
```sql
   WHERE [m].[Title] = N'action'
```
For performance reasons (caching the query plan by SQL Server) it's better to create parameterized queries such as:
```sql
   WHERE [m].[Title] =  @__p_0
```

**Comment by alirezanet** [comment]
Does this query translate to a parameterized Sql query in SqlServer ? 
```c#
dbContext.Blogs.Where( Param_0 =>  Param_0.Title == "action")
```
The thing is gridify doesn't control the provider's behavior. if you have a filter like `title=action` gridify generates the above query so you should see the same results in native LINQ and gridify generated expression, 
I will look into this more but first I need to find what is exactly the differences between these two expressions.

**Comment by alirezanet** [comment]
I did some research and few tests. the behavior of the Native LINQ and gridify generated expression is the same in SqlServer Provider. 

```c#
     [Fact]
      public void ApplyFiltering_GeneratedSqlShouldMatch_SqlServerProvider()
      {
         var actual = _dbContext.Users.ApplyFiltering("name = vahid").ToQueryString();
         var expected = _dbContext.Users.Where(q => q.Name == "vahid").ToQueryString();

         Assert.Equal(expected , actual);
      }
```
according to this test, the question is, does even this feature you mentioned exist in Entity Framework providers ?! if yes how we can use it because looks, like we can not use it even with Native LINQ., did I missed something?

**Comment by alirezanet** [comment]
Never mind, I found the parameterized way to do this in LINQ. 
This test is failing because of additional parameter exist in the final query:
```c#
      [Fact]
      public void ApplyFiltering_GeneratedSqlShouldMatch_UsingVariable_SqlServerProvider()
      {
         var name = "vahid";
         var actual = _dbContext.Users.ApplyFiltering("name = vahid").ToQueryString();
         var expected = _dbContext.Users.Where(q => q.Name == name).ToQueryString();

         Assert.Equal(expected , actual);
      }
```
I never encountered any best practices in the entity framework that says you should always define at least a variable for your constant values! Is this really matters !? (performance perspective)

**Comment by alirezanet** [comment]
This feature is available in version `v2.2.1`. 
To activate Query optimization for entity framework you need to use this method.
```c#
  GridifyGlobalConfiguration.EnableEntityFrameworkCompatibilityLayer();
```
